submarine-workbench/workbench-server/src/test/java/org/apache/submarine/server/WorkbenchGitHubServerTest.java

```diff
@@ -0,0 +1,266 @@
+    if (token == null) {
+      LOG.warn("Token not set!");
+      return;
+    } else {
+      Map<String, String> map = new HashMap<>();
+      map.put("Authorization", "token " + token);
+      LOG.info("gitToken: {}", token);
+      sendHttpRequest("https://api.github.com/repos/" + OWNER + "/" + REPO + "/subscription",
+          map, null, "PUT");
+
+      String result = sendHttpRequest("https://api.github.com/repos/" + OWNER + "/" + REPO
+          + "/subscribers", null, null, "GET");
+      LOG.info("Watching result: {}", result);
+
+      Gson gson = new Gson();
+      List<Map<String, String>> list = gson.fromJson(result, List.class);
+      assertEquals(1, list.size());
+      String login = list.get(0).get("login");
+      LOG.info("login: {}", login);
+      assertEquals(OWNER, login);
+    }
+  }
+
+  @Test
+  public void deleteWatching() {
+    String token = System.getenv("gitToken");
+    if (token == null) {
+      LOG.warn("Token not set!");
+      return;
+    } else {
+      Map<String, String> map = new HashMap<>();
+      map.put("Authorization", "token " + token);
+      LOG.info("gitToken: {}", token);
+      sendHttpRequest("https://api.github.com/repos/" + OWNER + "/" + REPO + "/subscription",
+          map, null, "PUT");
+
+      sendHttpRequest("https://api.github.com/repos/" + OWNER + "/" + REPO + "/subscription",
+          map, null, "DELETE");
+
+      String result = sendHttpRequest("https://api.github.com/repos/" + OWNER + "/" + REPO
+          + "/subscribers", null, null, "GET");
+      LOG.info("result: {}", result);
+
+      Gson gson = new Gson();
+      List<Map<String, String>> list = gson.fromJson(result, List.class);
+      assertEquals(0, list.size());
+    }
+  }
+
+  @Test
+  public void addStarring() {
+    String token = System.getenv("gitToken");
+    if (token == null) {
+      LOG.warn("Token not set!");
+      return;
+    } else {
+      Map<String, String> map = new HashMap<>();
+      map.put("Authorization", "token " + token);
+      LOG.info("gitToken: {}", token);
+      sendHttpRequest("https://api.github.com/user/starred/" + OWNER + "/" + REPO + "",
+          map, null, "PUT");
+
+      String result = sendHttpRequest("https://api.github.com/repos/" + OWNER + "/" + REPO
+          + "/stargazers", null, null, "GET");
+      LOG.info("Starring result: {}", result);
+
+      Gson gson = new Gson();
+      List<Map<String, String>> list = gson.fromJson(result, List.class);
+      assertEquals(1, list.size());
+      String login = list.get(0).get("login");
+      LOG.info("login: {}", login);
+      assertEquals(OWNER, login);
+    }
+  }
+
+  @Test
+  public void deleteStarring() {
+    String token = System.getenv("gitToken");
+    if (token == null) {
+      LOG.warn("Token not set!");
+      return;
+    } else {
+      Map<String, String> map = new HashMap<>();
+      map.put("Authorization", "token " + token);
+      LOG.info("gitToken: {}", token);
```

Comment:
For security reasons, token cannot be printed.

Reply:
Done.